plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/kernel/clients/KernelTableClient.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.kernel.clients;
+
+import io.delta.kernel.client.ExpressionHandler;
+import io.delta.kernel.client.FileSystemClient;
+import io.delta.kernel.client.JsonHandler;
+import io.delta.kernel.client.ParquetHandler;
+import io.delta.kernel.client.TableClient;
+import io.delta.kernel.defaults.client.DefaultExpressionHandler;
+import io.delta.kernel.defaults.client.DefaultFileSystemClient;
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.spi.type.TypeManager;
+import org.apache.hadoop.conf.Configuration;
+
+public class KernelTableClient
+        implements TableClient
+{
+    private final Configuration configuration;
+    private final TrinoFileSystem fileSystem;
+    private final TypeManager typeManager;
+
+    public KernelTableClient(
+            Configuration configuration,
```

Comment:
`trinodb/trino` has migrated off from `hadoop` related  API in favor of using natively the file system clients.
We'd likely not accept reintroducing this compile dependency in `trino-delta-lake`